hack/os-packages/Makefile

```diff
@@ -1,31 +1,37 @@
 
-artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_x86_64_fips_rpms.tar.gz:
-	cd artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el7_x86_64_fips_rpms && tar -czf ../os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_x86_64_fips_rpms.tar.gz *
+# fips enabled
+,PHONY: archive-os-packages.1
+archive-os-packages.1: artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_fips_rpms.tar.gz
+
+artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_rpms.tar.gz:
+	$(MAKE) download-all-rpms
+	cd artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_rpms && tar -czf ../os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_rpms.tar.gz *
+
+artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_fips_rpms.tar.gz:
+	$(MAKE) download-all-rpms
+	cd artifacts/os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_fips_rpms && tar -czf ../os-packages_$(DEFAULT_KUBERNETES_VERSION_SEMVER)_el$(rhel_major_version)_x86_64_fips_rpms.tar.gz *
 
 .PHONY: download-all-rpms
 download-all-rpms:
-	$(MAKE) rhel_major_version=7 fips=0 download-rpms
-	$(MAKE) rhel_major_version=7 fips=1 download-rpms
-#	$(MAKE) rhel_major_version=8 fips=0 bastion=0  download-rpms
-#	$(MAKE) rhel_major_version=7 fips=0 bastion=1  download-rpms
-#	$(MAKE) rhel_major_version=8 fips=0 bastion=1  download-rpms
-#	$(MAKE) rhel_major_version=7 fips=1 bastion=0  download-rpms
-#	$(MAKE) rhel_major_version=8 fips=1 bastion=0  download-rpms
+	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) download-rpms
 
 .PHONY: download-rpms
 download-rpms:
 	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) rpm-artifacts-getter.check
-	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) packages="audit ca-certificates conntrack-tools chrony curl ebtables open-vm-tools python3-pip python-netifaces python-requests socat sysstat yum-utils yum-plugin-versionlock nfs-utils NetworkManager" download-rpm
+	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) packages="audit ca-certificates conntrack-tools chrony curl ebtables open-vm-tools python3-pip python3-netifaces python3-requests socat sysstat yum-utils yum-plugin-versionlock nfs-utils NetworkManager" download-rpm
 # packages that are not in CentOS-7 Minimal
-	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) packages="tzdata nspr nss python-chardet yum-utils nss-softokn-freebl nss-util nss-softokn nss nss-tools  NetworkManager-libnm libdrm libpciaccess libxml2-python python-kitchen yum-plugin-ovl" download-rpm
+	$(MAKE) rhel_major_version=$(rhel_major_version) fips=$(fips) packages="tzdata nspr nss python3-chardet yum-utils nss-softokn-freebl nss-util nss-softokn nss nss-tools  NetworkManager-libnm libdrm libpciaccess python3-libxml2 python3-kitchen dnf-plugin-ovl" download-rpm
```

Comment:
python-netifaces, python-requests, python-chardet, libxml2-python  packages were renamed to `python3` in centos 8 and not available on centos 8 epel repos. while `python3-*` are available in both centos 7 and 8. 

I noticed we already use `python3-pip` for centos 7, so changing to `python3-*` should not break (theoretically) centos 7 installations. I have to run full end to end test on all flavors to verify that though. 

otherwise we have to maintain separate list of rpms for each major OS versions

Reply:
refactored makefile to download OS version specific packages